src/platform/packages/shared/kbn-unified-tabs/src/components/tab/tab.tsx

```diff
@@ -15,13 +15,47 @@ import {
   EuiFlexGroup,
   EuiFlexItem,
   EuiText,
-  EuiThemeComputed,
   useEuiTheme,
+  type EuiThemeComputed,
 } from '@elastic/eui';
 import { TabMenu } from '../tab_menu';
 import { EditTabLabel, type EditTabLabelProps } from './edit_tab_label';
 import { getTabAttributes } from '../../utils/get_tab_attributes';
-import type { TabItem, TabsSizeConfig, GetTabMenuItems } from '../../types';
+import type { TabItem, TabsSizeConfig, GetTabMenuItems, PreviewContentConfig } from '../../types';
+import { TabPreview } from '../tab_preview';
+
+// TODO: replace with real data when ready
+const TAB_CONTENT_MOCK: PreviewContentConfig[] = [
```

Comment:
Let's make it easier to consumer tabs package by removing the mock from it. We can still use such mocked data for now but define it on a consumer side. Currently it would be in the UnifiedTabs example plugin https://github.com/elastic/kibana/blob/5711bb9d90680483e215b8edc7da3942acfdf422/examples/unified_tabs_examples/public/example_app.tsx#L99

For example, we can introduce a prop `getTabPreviewData: (item: TabItem) => (type with { query, status } structure)` and pass it via `TabsBar` down to `TabPreview` component.